gateway/handlers/reader.go

```diff
@@ -65,3 +78,36 @@ func MakeFunctionReader(metricsOptions metrics.MetricOptions, c *client.Client)
 		w.Write(functionBytes)
 	}
 }
+
+func getReplicaInfo(c *client.Client, services []swarm.Service, ctx context.Context) map[string]int {
+	// Begin replica info section
+	taskFilter := filters.NewArgs()
+
+	taskFilter.Add("desired-state", "running")
```

Comment:
Not sure if you are filtering the services here?

Reply:
I need to filter the tasks list instead of the services list. In docker swarm API, the Services list takes a different set of filters. I don't think they have a filter that will help us get running state:

https://github.com/docker/swarmkit/blob/da5ee2a69176ca3c1eed1b04c4308bc1ab932c05/api/control.pb.go#L298